File: ts/indicators/john-ehlers/hilbert-transformer/cycle-estimator-type.ts
/** Enumerates types of techniques to estimate an instantaneous period using a Hilbert transformer. */
export enum HilbertTransformerCycleEstimatorType {

  /** An instantaneous period estimation based on the homodyne discriminator technique. */
  HomodyneDiscriminator,

  /** An instantaneous period estimation based on the homodyne discriminator technique
   * (TA-Lib implementation with unrolled loops).
   */
  HomodyneDiscriminatorUnrolled,

  /** An instantaneous period estimation based on the phase accumulation technique. */
  PhaseAccumulator,

  /** An instantaneous period estimation based on the dual differentiation technique. */
  DualDifferentiator
}

File: py/indicators/john_ehlers/hilbert_transformer/cycle_estimator_type.py
"""Cycle estimator type enumeration."""
from enum import IntEnum


class CycleEstimatorType(IntEnum):
    """Enumerates types of techniques to estimate an instantaneous period
    using a Hilbert transformer."""

    HOMODYNE_DISCRIMINATOR = 0
    HOMODYNE_DISCRIMINATOR_UNROLLED = 1
    PHASE_ACCUMULATOR = 2
    DUAL_DIFFERENTIATOR = 3
